soc/xtensa/intel_adsp/common/soc_mp.c

```diff
@@ -67,23 +67,44 @@ struct cpustart_rec {
 	uint32_t		vecbase;
 
 	uint32_t		alive;
-
-	/* padding to cache line */
-	uint8_t		padding[XCHAL_DCACHE_LINESIZE - 6 * 4];
 };
```

Comment:
...and this commit (again, the complete commit) fixes the previously introduced breakage again.

Reply:
This is the "Clean up cache handling in MP startup" patch?  There was one flush in the early code that should have been an invalidate, which I guess might explain that if this is what you were seeing.  The change to inlines in the cache code that you mentioned above shouldn't change behavior, but it will change the amount of stack used and thus the data flushed.

Reply:
@andyross the one, fixing the breakage - yes, and the one, introducing the breakage is "Add non-HAL caching primitives"